geoportailv3/static/js/query/querydirective.js

```diff
@@ -0,0 +1,242 @@
+goog.provide('app.queryDirective');
+
+
+/**
+ * @param {string} appQueryTemplateUrl URL to directive template.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+app.queryDirective = function(appQueryTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appQueryMap',
+      'infoOpen': '=appQueryOpen'
+    },
+    controller: 'AppQueryController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appQueryTemplateUrl
+  };
+};
+app.module.directive('appQuery', app.queryDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.$http} $http Angular $http service
+ * @param {string} appQueryTemplatesPath Path
+ *                 to find the intterogation templates.
+ * @param {string} getInfoServiceUrl
+ * @param {string} getRemoteTemplateServiceUrl
+ * @export
+ * @ngInject
+ */
+app.QueryController = function($scope, $http,
+    appQueryTemplatesPath, getInfoServiceUrl,
+    getRemoteTemplateServiceUrl) {
+
+  this['content'] = [];
```

Comment:
It would be good to provide a type for this. For example, it makes the code easier to understand to others.